src/layer/construction.js

```diff
@@ -0,0 +1,91 @@
+const majorConstruction = [
+  "any",
+  ["in", "motorway", ["get", "class"]],
+  ["in", "trunk", ["get", "class"]],
+];
+
+const constructionColor = [
+  "interpolate-lab",
+  ["exponential", 2],
+  ["zoom"],
+  10,
+  ["case", majorConstruction, "lightcoral", "lightslategray"],
+  13,
+  ["case", majorConstruction, "maroon", "lightslategray"],
+  15,
+  ["case", majorConstruction, "maroon", "slategray"],
+];
+
+const constructionFilter = [
+  "in",
+  ["get", "class"],
+  [
+    "literal",
+    [
+      "motorway_construction",
+      "trunk_construction",
+      "primary_construction",
+      "secondary_construction",
+      "tertiary_construction",
+      "minor_construction",
+      "service_construction",
+    ],
+  ],
+];
+
+export const road = {
+  id: "highway-construction",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: constructionFilter,
+  minzoom: 9,
+  paint: {
+    "line-color": constructionColor,
+    "line-opacity": ["interpolate", ["exponential", 2], ["zoom"], 10, 0, 11, 1],
+    "line-width": [
+      "interpolate",
+      ["exponential", 2],
+      ["zoom"],
+      9,
+      0,
+      12,
+      ["case", majorConstruction, 1, 0.5],
+      15,
+      1,
+    ],
+    "line-dasharray": [2, 1],
+    "line-offset": 0,
+    "line-gap-width": [
+      "interpolate",
+      ["exponential", 2],
+      ["zoom"],
+      11,
+      0,
+      20,
+      2,
+    ],
```

Comment:
Good idea to make everything thin, including the dashes. That neatly gets around the issues with overlapping, mismatched dashes that I had run into in #215. However, the dashing can be a very subtle effect at some zoom levels. In the screenshot below, the U-shaped road is also under construction:

[<img src="https://user-images.githubusercontent.com/1231218/179844182-ad075b2b-6ab0-436a-9752-0f195808bb46.png" width="400" alt="Vallée-des-Forts">](https://zelonewolf.github.io/openstreetmap-americana/#14/45.09346/-73.08425)

Can we widen the gaps a bit at the lower zoom levels to keep the roads from looking like a different classification of corporeal road? It seems to b more of an issue with non-motorways; the blue contrasts less well with the land background color.

We should also double-check the output on 1× displays.

Reply:
Changed line-width to 1, always, causes the dashes and gaps to be uniform across both stylings:

<img width="256" alt="image" src="https://user-images.githubusercontent.com/23022/179846982-016c33f5-4ba4-42b1-91fd-0b8d43322c18.png">